packages/utils-dom/src/htmldocument.ts

```diff
@@ -165,8 +165,17 @@ export class HTMLDocument extends Node {
     /**
      * https://developer.mozilla.org/docs/Web/API/Document/querySelector
      */
-    public querySelector(selector: string): HTMLElement {
-        return this.querySelectorAll(selector)[0];
+    public querySelector(selector: string): HTMLElement | null {
+        if (!CACHED_CSS_SELECTORS.has(selector)) {
+            CACHED_CSS_SELECTORS.set(selector, cssSelect.compile(selector));
+        }
+
+        const data: any | undefined = cssSelect.selectOne(
+            CACHED_CSS_SELECTORS.get(selector) as cssSelect.CompiledQuery,
```

Comment:
Can't the cache cause troubles if we switch to analyze continuously?

Reply:
Good question,

It will not cause any issues because "compiled selectors" are unique per css selector and have nothing to do with the DOM, so '#my-id div.my-class' will always _compile_ to the same function.

For more details see: https://github.com/fb55/css-select#how-does-it-work

Additionally in the current Edge DevTools implementation, we get a fresh webhint instance per DOM snapshot when running continuously.